cli/clistat/stat.go

```diff
@@ -0,0 +1,218 @@
+package clistat
+
+import (
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-sysinfo"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+	"tailscale.com/types/ptr"
+
+	sysinfotypes "github.com/elastic/go-sysinfo/types"
+)
+
+// Prefix is an SI prefix for a unit.
+type Prefix string
+
+// Float64 returns the prefix as a float64.
+func (m *Prefix) Float64() (float64, error) {
+	switch *m {
+	case PrefixDeciShort, PrefixDeci:
+		return 0.1, nil
+	case PrefixCentiShort, PrefixCenti:
+		return 0.01, nil
+	case PrefixMilliShort, PrefixMilli:
+		return 0.001, nil
+	case PrefixMicroShort, PrefixMicro:
+		return 0.000_001, nil
+	case PrefixNanoShort, PrefixNano:
+		return 0.000_000_001, nil
+	case PrefixKiloShort, PrefixKilo:
+		return 1_000.0, nil
+	case PrefixMegaShort, PrefixMega:
+		return 1_000_000.0, nil
+	case PrefixGigaShort, PrefixGiga:
+		return 1_000_000_000.0, nil
+	case PrefixTeraShort, PrefixTera:
+		return 1_000_000_000_000.0, nil
+	case PrefixKibiShort, PrefixKibi:
+		return 1024.0, nil
+	case PrefixMebiShort, PrefixMebi:
+		return 1_048_576.0, nil
+	case PrefixGibiShort, PrefixGibi:
+		return 1_073_741_824.0, nil
+	case PrefixTebiShort, PrefixTebi:
+		return 1_099_511_627_776.0, nil
+	default:
+		return 0, xerrors.Errorf("unknown prefix: %s", *m)
+	}
+}
+
+const (
+	PrefixDeci  Prefix = "deci"
+	PrefixCenti Prefix = "centi"
+	PrefixMilli Prefix = "milli"
+	PrefixMicro Prefix = "micro"
+	PrefixNano  Prefix = "nano"
+
+	PrefixDeciShort  Prefix = "d"
+	PrefixCentiShort Prefix = "c"
+	PrefixMilliShort Prefix = "m"
+	PrefixMicroShort Prefix = "u"
+	PrefixNanoShort  Prefix = "n"
+
+	PrefixKilo Prefix = "kilo"
+	PrefixMega Prefix = "mega"
+	PrefixGiga Prefix = "giga"
+	PrefixTera Prefix = "tera"
+
+	PrefixKiloShort Prefix = "K"
+	PrefixMegaShort Prefix = "M"
+	PrefixGigaShort Prefix = "G"
+	PrefixTeraShort Prefix = "T"
+
+	PrefixKibi = "kibi"
+	PrefixMebi = "mebi"
+	PrefixGibi = "gibi"
+	PrefixTebi = "tebi"
+
+	PrefixKibiShort Prefix = "Ki"
+	PrefixMebiShort Prefix = "Mi"
+	PrefixGibiShort Prefix = "Gi"
+	PrefixTebiShort Prefix = "Ti"
+)
```

Comment:
Surprised there isn't a good unit parsing library that could handle this logic.

Reply:
Kubernetes apimachinery does have one but it's pretty heavyweight IIRC.